src/test/csf/timers.h

```diff
@@ -30,6 +30,139 @@ namespace csf {
 
 // Timers are classes that schedule repeated events and are mostly independent
 // of simulation-specific details.
+/*
+    We should consider making DurationDistribution conform to
+    RandomNumberDistribution concept and moving to beast.
```

Comment:
What do you mean by move to beast? Is there something there we can reuse?

Reply:
When I was originally constructing the DurationDistribution concept, I had it as templated, such that the object returned by `operator()` could be of any time. The idea was to have a concept:

```
template <class T>
Distribution
{
  T
  operator();
};
```

that could represent a random distribution of any object. But I didn't go through with that level of genericness in order to focus on getting things working.

Reply:
I do think, however, that the idea of having the concept of a random (the precise sense in which it is random is not too important) distribution, that any type of object can be sampled from is very powerful and worth generalizing either in beast or in the C++ standard.

But I'm not expert and I didn't look very hard so It's entirely possible that something like that already exists and I just didn't look hard enough.